File: ScheduleView/DatabaseEntityFramework/RecurrenceHelper.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Telerik.Windows.Controls.ScheduleView;
using Telerik.Windows.Controls.ScheduleView.ICalendar;

namespace DatabaseEntityFramework
{
    public class RecurrenceHelper
    {
        public static bool IsOccurrenceInRange(string valueToParse, DateTime start, DateTime end)
        {
            RecurrencePattern pattern = new RecurrencePattern();
            if (RecurrencePatternHelper.TryParseRecurrencePattern(valueToParse, out pattern))
            {
                return pattern.GetOccurrences(start, start, end).Count() > 0;
            }

            return false;
        }
    }
}

File: ScheduleView/Database/WPF_VB/Helpers/RecurrenceHelper.vb
Imports Telerik.Windows.Controls.ScheduleView
Imports Telerik.Windows.Controls.ScheduleView.ICalendar

Public Class RecurrenceHelper
	Public Shared Function IsOccurrenceInRange(valueToParse As String, start As DateTime, [end] As DateTime) As Boolean
		Dim pattern As New RecurrencePattern()
		If RecurrencePatternHelper.TryParseRecurrencePattern(valueToParse, pattern) Then
			Return pattern.GetOccurrences(start, start, [end]).Count() > 0
		End If

		Return False
	End Function
End Class
